contracts/SCBridgeWallet.sol

```diff
@@ -174,6 +200,17 @@ contract SCBridgeWallet is IAccount {
     // If the function is permitted, it can be called at any time
     // (including when the wallet is in CHALLENGE_RAISED) with no futher checks.
     bytes4 functionSelector = bytes4(userOp.callData[0:4]);
+
+    if (functionSelector == this.crossChainExecute.selector) {
+      validateCrossChainExecute(userOp.callData[4:]);
+    } else {
+      // Prevent replay attacks by checking against the entrypoint and chain id
+      require(
+        userOpHash == generateUserOpHash(userOp, entrypoint, block.chainid),
```

Comment:
This prevents replay attacks, since this check will fail if the UserOp is submitted to a different entrypoint or chain.